File: Server/Resources/Asset/AssetFilter.cs
using System;

namespace Server
{
    public class AssetFilter : Filter
    {
        public int[] Ids { get; set; }
        public Nullable<DateTime> PurchaseDateMin { get; set; }
        public Nullable<DateTime> PurchaseDateMax { get; set; }
        public ProductFilter ProductFilter { get; set; }
        public OfficeFilter OfficeFilter { get; set; }
    }
}

File: Client/src/app/shared/resources/asset/asset.filter.ts
import { OfficeFilter } from "../office/office.filter";
import { ProductFilter } from "../product/product.filter";

export class AssetFilter {
    ids?: number[];
    purchaseDateMin?: Date;
    purchaseDateMax?: Date;
    productFilter?: ProductFilter;
    officeFilter?: OfficeFilter;
}